src/appleseed/renderer/modeling/bssrdf/normalizeddiffusionbssrdf.cpp

```diff
@@ -136,8 +170,15 @@ namespace
             sample.set_is_directional(false);
             sample.set_eta(values->m_inside_ior / values->m_outside_ior);
 
-            // Select the channel leading to the strongest scattering.
-            const size_t channel = min_index(values->m_reflectance);
+            sample.get_sampling_context().split_in_place(3, 1);
+            const Vector3d s = sample.get_sampling_context().next_vector2<3>();
+
+            // Sample a channel.
+            const size_t channel =
+                sample_cdf(
+                    values->m_channel_cdf.begin(),
+                    values->m_channel_cdf.end(),
+                    s[0]);
             sample.set_channel(channel);
 
             // todo: fix.
```

Comment:
The fix will be to make sure we never select a channel for which reflectance is 0.